File: Dialogs Source Code/VideoEffects/IVideoEffectsSettingsDialog.cs
// --------------------------------------------------------------------------------------------------------------------
// <copyright file="IVideoEffectsSettingsDialog.cs" company="VisioForge">
//   VisioForge (c) 2006 - 2021
// </copyright>
// <summary>
//   IVideoEffectsSettingsDialog interface.
// </summary>
// --------------------------------------------------------------------------------------------------------------------

namespace VisioForge.Core.UI.WinForms.Dialogs.VideoEffects
{
    using VisioForge.Core.MediaPlayer;
    using VisioForge.Core.VideoCapture;
    using VisioForge.Core.VideoEdit;
    using VisioForge.Core.Types.VideoEffects;

    /// <summary>
    /// IVideoEffectsSettingsDialog interface.
    /// </summary>
    public interface IVideoEffectsSettingsDialog
    {
        /// <summary>
        /// Attaches effect.
        /// </summary>
        /// <param name="effect">
        /// Effect.
        /// </param>
        void Attach(IVideoEffect effect);

        /// <summary>
        /// Fills effect.
        /// </summary>
        /// <param name="effect">
        /// Effect.
        /// </param>
        void Fill(IVideoEffect effect);

        /// <summary>
        /// Generates new effect name.
        /// </summary>
        /// <param name="core">
        /// Core.
        /// </param>
        /// <returns>
        /// The <see cref="string"/>.
        /// </returns>
        string GenerateNewEffectName(IVideoEditCore core);

        /// <summary>
        /// Generates new effect name.
        /// </summary>
        /// <param name="core">
        /// Core.
        /// </param>
        /// <returns>
        /// The <see cref="string"/>.
        /// </returns>
        string GenerateNewEffectName(IVideoCaptureCore core);

        /// <summary>
        /// Generates new effect name.
        /// </summary>
        /// <param name="core">
        /// Core.
        /// </param>
        /// <returns>
        /// The <see cref="string"/>.
        /// </returns>
        string GenerateNewEffectName(IMediaPlayerCore core);
    }
}

File: Dialogs Source Code/VideoEffects/IVideoEffectsSettingsDialog.vb
' --------------------------------------------------------------------------------------------------------------------
' <copyright file="IVideoEffectsSettingsDialog.vb" company="VisioForge">
'   VisioForge (c) 2006 - 2021
' </copyright>
' <summary>
'   IVideoEffectsSettingsDialog interface.
' </summary>
' --------------------------------------------------------------------------------------------------------------------

Namespace VisioForge.Core.UI.WinForms.Dialogs.VideoEffects
    Imports VisioForge.Core.MediaPlayer
    Imports VisioForge.Core.VideoCapture
    Imports VisioForge.Core.VideoEdit
    Imports VisioForge.Core.Types.VideoEffects

    ''' <summary>
    ''' IVideoEffectsSettingsDialog interface.
    ''' </summary>
    Public Interface IVideoEffectsSettingsDialog
        ''' <summary>
        ''' Attaches effect.
        ''' </summary>
        ''' <param name="effect">
        ''' Effect.
        ''' </param>
        Sub Attach(effect As IVideoEffect)

        ''' <summary>
        ''' Fills effect.
        ''' </summary>
        ''' <param name="effect">
        ''' Effect.
        ''' </param>
        Sub Fill(effect As IVideoEffect)

        ''' <summary>
        ''' Generates new effect name.
        ''' </summary>
        ''' <param name="core">
        ''' Core.
        ''' </param>
        ''' <returns>
        ''' The <see cref="String"/>.
        ''' </returns>
        Function GenerateNewEffectName(core As IVideoEditCore) As String

        ''' <summary>
        ''' Generates new effect name.
        ''' </summary>
        ''' <param name="core">
        ''' Core.
        ''' </param>
        ''' <returns>
        ''' The <see cref="String"/>.
        ''' </returns>
        Function GenerateNewEffectName(core As IVideoCaptureCore) As String

        ''' <summary>
        ''' Generates new effect name.
        ''' </summary>
        ''' <param name="core">
        ''' Core.
        ''' </param>
        ''' <returns>
        ''' The <see cref="String"/>.
        ''' </returns>
        Function GenerateNewEffectName(core As IMediaPlayerCore) As String
    End Interface
End Namespace
